src/utilities/resolveSchemaCoordinate.js

```diff
@@ -0,0 +1,111 @@
+import type { GraphQLSchema } from '../type/schema';
+import type { GraphQLSchemaElement } from '../type/element';
+import type { SchemaCoordinateNode } from '../language/ast';
+import type { Source } from '../language/source';
+import {
+  isObjectType,
+  isInterfaceType,
+  isEnumType,
+  isInputObjectType,
+} from '../type/definition';
+import { parseSchemaCoordinate } from '../language/parser';
+
+/**
+ * A schema coordinate is resolved in the context of a GraphQL schema to
+ * uniquely identifies a schema element. It returns undefined if the schema
+ * coordinate does not resolve to a schema element.
+ *
+ * https://spec.graphql.org/draft/#sec-Schema-Coordinates.Semantics
+ */
+export function resolveSchemaCoordinate(
+  schema: GraphQLSchema,
+  schemaCoordinate: string | Source,
+): GraphQLSchemaElement | void {
```

Comment:
From a DX standpoint returning just an object is bad because fields, args, and enums values are plain objects. That means you can easily detect what was returned.

There two solutions wrap the result in object and add `kind` tag field, e.g.:
```
{
  kind: Kind.ARGUMENT,
  name: 'someArg',
  ...
}
```
alternatively, it can be, "one of" pattern:

```
{
  arguments: {
    name: 'someArg',
    // ...
  },
}
```